plugins/cue_points/base/lib/kCuePointManager.php

```diff
@@ -249,6 +260,32 @@ public function objectReplaced(BaseObject $object, BaseObject $replacingObject,
 		return true;
 	}
 
+	/* (non-PHPdoc)
+	 * @see kObjectCopiedEventConsumer::objectCopied()
+	 */
+	public function objectCopied(BaseObject $fromObject, BaseObject $toObject)
+	{
+		if($fromObject instanceof entry)
+		{
+			$c = new KalturaCriteria();
+			$c->add( CuePointPeer::ENTRY_ID, $fromObject->getId() );
+			$c->addAscendingOrderByColumn(CuePointPeer::CREATED_AT);
+			$cpToCopy = CuePointPeer::doSelect($c);
+			if ( count( $cpToCopy ) <= self::MAX_CUE_POINTS_TO_COPY )
```

Comment:
You better call all cue-point plugins to check if their copy is enabled and select only the cue-points of these types, instead of selecting only 1000 and checking them one by one.